Safaa/src/safaa/Safaa.py

```diff
@@ -241,12 +256,13 @@ def train_false_positive_detector_model(self, data, labels):
         # Preprocess the data before training
         preprocessed_data = self.preprocess_data(data)
         # Fit the vectorizer to the preprocessed data
-        vectorized_data = self.vectorizer.fit_transform(preprocessed_data)
+        vectorized_data = self.vectorizer.transform(preprocessed_data)
```

Comment:
Changing from 
`vectorized_data = self.vectorizer.fit_transform(preprocessed_data)`
to
`vectorized_data = self.vectorizer.transform(preprocessed_data)`
makes sense now to align the code with incremental learning. The main problem is vectorizers don't support this feature, so the pipeline needs to take this point into consideration, unless we don't plan on updating the corpus often.

Maybe adding a separate function for training the vectorizer from scratch might be of benefit, where the pipeline can accumulate some data first then it can be trained on that entire combined corpus every once in a while.

Not necessarily a request to change anything. Just something to keep in mind.

Reply:
Yes I also found that issue yesterday, when I was shifting. I think we need to create a stage for vectorizer training as well. Which we might need to think about :)
How we can keep the track of the data :'( is the issue we need to resolve.